beps/0012-default-permission-decisions/README.md

```diff
@@ -0,0 +1,190 @@
+  - [API Changes](#api-changes)
+- [Release Plan](#release-plan)
+- [Alternatives](#alternatives)
+  - [Return Default Decisions from Permission Backend](#return-default-decisions-from-the-permission-backend)
+  - [Communicate Default Decisions Over Separate Channel](#communicate-default-decisions-over-separate-channel)
+
+## Summary
+
+The Backstage Permission System currently requires explicit policy handlers to be defined for all permissions. This BEP proposes adding support for default permission decisions that can be defined when creating a permission. These default decisions can be either definitive or conditional, leveraging existing evaluation mechanisms. This enhancement enables plugins to define built-in behavior through the permission system while maintaining the flexibility for adopters to override this behavior with custom policies.
+
+## Motivation
+
+The current permission system requires explicit policy handlers for all permissions, unless the permission system is disabled in which case all permissions are allowed. There have been discussions in the past about supporting partial or default permission policies provided by plugins, see [#13891](https://github.com/backstage/backstage/issues/13891) and [#18064](https://github.com/backstage/backstage/pull/18064). Being able to provide default policies is useful in many cases, for example, a plugin might want to only allow access to owners of a resource, but allow a policy author to override this for specific permissions. The inverse case is also important, where a plugin wants to expose sensitive functionality that should only be available to a restricted set of users, but be unavailable by default. The lack of this capability essentially prevents implementation of these kind of features in a clean way, forcing plugin authors to resort to custom mechanisms.
+
+### Goals
+
+- Enable plugin authors to define default policies for their permissions
+  - Support both basic and resource permissions
+  - Support for conditional decisions based on user information
+  - Have these policies remain in effect even if the permission system is disabled
+- Maintain backward compatibility with existing permission system
+- Improve security by enabling secure-by-default patterns
+- Allow permission policy authors to defer to the default decision
+
+### Non-Goals
+
+- Configuration of default decisions
+- Complex business logic in default decisions
+
+## Proposal
+
+A new "default" authorize result is added to the permission system that can be returned by a policy handler to indicate that the permission should be evaluated using the default decision. The permission system supports two variants of default decisions: "default or allow" and "default or deny". These indicate that the permission should be evaluated using the default decision, and if no default decision is defined, it will fall back to "allow" or "deny" respectively.
```

Comment:
An alternative to simplify this a bit could be to add an utility function that detects whether a permission has default decision:

```ts
  if(hasDefaultDecision(permission)) {
    return DefaultPolicyDecision;
  } 
  return DENY;
```

this way adopters could build "default or allow" or "default or deny"

Reply:
Hmm, kinda tricky to make that information available at that point thought right? Especially considering https://github.com/backstage/backstage/pull/29578#discussion_r2056684747

Reply:
it depends if the permission framework can discover the information somehow